src/common/models/essence/essence.ts

```diff
@@ -67,6 +65,19 @@ export enum VisStrategy {
   KeepAlways
 }
 
+type MeasureId = string;
+type DimensionId = string;
+
+interface MeasuresDefinition {
```

Comment:
I don't really like this interface, it is very easy to create invalid object, there is also duplication in the code with conditionals based on isMulti.

Reply:
Valid points, but on the other hand, we now have a lot of cases in code, where we depend on availability of multi and single fields. It holds both selections and restores old values when switching. So union of `SingleMeasure` and `MultiMeasure` won't do. I don't have better ideas but will appreciate help there.